### PR TITLE
Add space connected apps doc

### PR DESCRIPTION
**Pull request by coyotte508**
cc @Wauplin @gary149 @julien-c

### CODE REVIEW

File: docs/hub/oauth.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Contact us at [website@huggingface.co](mailto:website@huggingface.co) with the desired name, logo url, scope and redirect urls for the OAuth app. We will provide you with the client ID and client secret.
+
+All other information is available in the [OpenID metadata](https://huggingface.co/.well-known/openid-configuration).
+
+## Branding
+
+You are free to use the Hugging Face logo in the "Sign in with Hugging Face" button.
+
+Here's an example button in HTML+CSS:
+
```
Comment (Wauplin) [inline docs/hub/oauth.md:16]: Maybe add an example of how the snippet of code renders on a website? Even if it's just an image I think it's nice to have an idea of what to expect.

File: docs/hub/spaces-oauth.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Adding a Sign-In with HF button to your space
+
+It's possible to create an OAuth / OpenID connect app associated to your space. This will allow users to sign-in to your space using their HF account.
```
Comment (Wauplin) [inline docs/hub/spaces-oauth.md:3]: I would add a link to a guide/tutorial/article explaining the basics of OAuth

File: docs/hub/spaces-oauth.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+Basically, you need to:
+
+- Redirect the user to `https://huggingface.co/oauth/authorize?redirect_uri={REDIRECT_URI}&scope=openid%20profile&client_id={CLIENT_ID}&state={STATE}`, where `STATE` is a random string that you will need to verify later.
+- Handle the callback on `/auth/callback` or `/login/callback` (depending on the type of app you created) and verify the `state` parameter.
+- Use the `code` query parameter to get an access token and id token from `https://huggingface.co/oauth/token` (POST request with `client_id`, `code`, `grant_type=authorization_code` and `redirect_uri` as form data, and with `Authorization: Basic {base64(client_id:client_secret)}` as a header)
+
+<Tip warning={true}>
+
+If the space is running inside an iframe, you should use `target=_blank` on the button to open the sign-in page in a new tab. Otherwise, you might encounter issues with the cookies on some browsers.
```
Comment (Wauplin) [inline docs/hub/spaces-oauth.md:69]: I would rephrase it since all spaces are running in an iframe "by default"

**Review by Wauplin — COMMENTED** [review]
Nice guide. I added a few comments but overall the content is there IMO :)

File: docs/hub/oauth.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Sign in with Hugging Face
+
+Besides [spaces](https://huggingface.co/docs/hub/spaces-oauth), it's also possible to create an OAuth / OpenID connect app for your website. This will allow users to sign-in to your website using their HF account.
+
+It's also possible to have the additional `email` scope.
```
[the remark this answers is not in the document]
Comment (osanseviero) [reply docs/hub/oauth.md:5]: Agree + a short explanation of each

Comment (osanseviero) [inline docs/hub/oauth.md:5]: I would maybe add a small explanation that we don't provide access to write repos for example, which will be one of the most common requested use cases

**Review by osanseviero — COMMENTED** [review]
Very cool! Looks nice

File: docs/hub/oauth.md (hunk 1 above)
Comment (gary149) [reply docs/hub/oauth.md:20]: Instead of complex CSS, you can tell to use a badge maybe (in a <a>)?
https://huggingface.co/datasets/huggingface/badges#sign-in-with-hugging-face

File: docs/hub/spaces-oauth.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
+# Adding a Sign-In with HF button to your Space
+
+You can enable a built-in sign-in flow to your Space so users can log in with their HF account. To do this, you can create an [OAuth/OpenID connect](https://developer.okta.com/blog/2019/10/21/illustrated-guide-to-oauth-and-oidc) app associated to your Space.
```
[the remark this answers is not in the document]
Comment (Pierrci) [reply docs/hub/spaces-oauth.md]: I pushed some tweaks and I added https://github.com/huggingface/hub-docs/pull/902/files#diff-8c77839fc5d40b69830303ec9567fbc0a4fbbefabf3309abef7de83f758f2652R5

File: docs/hub/spaces-oauth.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+The following scopes are available:
+
+- `openid`: Get the ID token in addition to the access token.
+- `profile`: Get the user's profile information (username, avatar, etc.)
+
+You should use `"openid profile"` as the scope for your OAuth app.
+
+## Adding the button to your Space
+
+You now have all the information to add a "Sign-in with HF" button to your Space. Libraries in [various](https://github.com/lepture/authlib) [languages](https://github.com/panva/node-openid-client) can help you implement the OpenID/OAuth protocol. Gradio also provides **built-in support**, making implementing the Sign-in with HF button a breeze; you can [check out the associated doc](TODO).
```
Comment (Pierrci) [inline docs/hub/spaces-oauth.md:60]: ping @Wauplin regarding the `TODO`, not sure there's a doc for Gradio support already or not? feel free to push changes directly!

**Comment by Wauplin** [comment]
@Pierrci I pushed a few tweaks in https://github.com/huggingface/hub-docs/pull/902/commits/434c3b4fda7f54474ed2083fed36717b9210160e. IMO, we are good to go with this PR

File: docs/hub/spaces-oauth.md (hunk 6 above)
Comment (Wauplin) [reply docs/hub/spaces-oauth.md]: There's a small guide yes. See https://github.com/huggingface/hub-docs/commit/434c3b4fda7f54474ed2083fed36717b9210160e

**Comment by Pierrci** [comment]
Thanks for the twists @Wauplin, looks great! I think we'll merge this once https://github.com/huggingface/moon-landing/pull/7246 is deployed.

**Review by Pierrci — APPROVED** [review]

**Comment by julien-c** [comment]
late to the party but the https://huggingface.co/docs/hub/oauth page in particular i feel like is missing a bit of context/visuals, so opened a follow-up PR at #942
